crates/consensus/debug-client/src/client.rs

```diff
@@ -0,0 +1,207 @@
+use alloy_consensus::TxEnvelope;
+use alloy_eips::eip2718::Encodable2718;
+use reth_node_api::EngineTypes;
+use reth_node_core::{
+    primitives::B256,
+    rpc::types::{BlockTransactions, ExecutionPayloadV2, ExecutionPayloadV3, RichBlock},
+};
+use reth_rpc_builder::auth::AuthServerHandle;
+use reth_rpc_types::ExecutionPayloadV1;
+use ringbuffer::{AllocRingBuffer, RingBuffer};
+use std::{future::Future, sync::Arc};
+use tokio::sync::mpsc;
+
+/// Supplies consensus client with new blocks sent in `tx` and a callback to find specific blocks
+/// by number to fetch past finalized and safe blocks.
+pub trait BlockProvider: Send + Sync + 'static {
+    /// Spawn a block provider to send new blocks to the given sender.
+    fn spawn(&self, tx: mpsc::Sender<RichBlock>) -> impl Future<Output = ()> + Send;
```

Comment:
What would you prefer, this or `impl Stream<Output = RichBlock>`? IMO `impl Stream<Output = RichBlock>` is much less nicer to implement (bc of manual state machine building vs just `async fn`), but can change it if you prefer